### DISCUSSION TITLE
Compilation error with MSVC 2019, cmake, windows 11

### DISCUSSION THREAD

**Discussion by stef-pellegrino** [ccache/ccache#1027] [Support]
Cannot use ccache on Windows 11, QtCreator 5.0.2, cmake   3.23.0-rc1, MSVC 2019 (32bit)

Add
```
find_program(CCACHE_PROGRAM ccache)

if (CCACHE_PROGRAM)
    message(NOTICE "ccache is enabled (found here: ${CCACHE_PROGRAM})")
    set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "\"${CCACHE_PROGRAM}\"")
    set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK "\"${CCACHE_PROGRAM}\"")
else ()
    message(FATAL_ERROR "ccache has not been found")
endif ()

```

I get this message when compiling :

`-1: warning: D9002 : ignoring unknown option '/fo'`
`c1 : error: C1083: Cannot open source file: 'proj\CMakeFiles\proj.dir\version.rc.res.dep.obj ': No such file or directory`

**Comment by cristianadam** [comment]
Do not use `RULE_LAUNCH_COMPILE` and `RULE_LAUNCH_LINK`. This way of using `ccache` with `cmake` is obsolete.

Try with:

```cmake
find_program(CCACHE_PROGRAM ccache)

if (CCACHE_PROGRAM)
    message(NOTICE "ccache is enabled (found here: ${CCACHE_PROGRAM})")
    set(CMAKE_CXX_COMPILER_LAUNCHER "${CCACHE_PROGRAM}" CACHE STRING "CXX compiler launcher" FORCE)
    set(CMAKE_C_COMPILER_LAUNCHER "${CCACHE_PROGRAM}" CACHE STRING "C compiler launcher" FORCE)
else ()
    message(FATAL_ERROR "ccache has not been found")
endif ()
```
instead.

**Reply by cristianadam** [reply]
Qt Creator itself is being built using `ccache` at GitHub Actions at https://github.com/qt-creator/qt-creator/actions 
MSVC 2019 (x64) + ninja is being used and works fine.

**Reply by stef-pellegrino** [reply]
Ok it works, Thanks for your help !!!
